src/VolleyManagement.Backend/VolleyManagement.UI/Areas/Mvc/Controllers/Domain/TeamsController.cs

```diff
@@ -103,17 +103,13 @@ public JsonResult Create(TeamViewModel teamViewModel)
 
                     var players = _playerService.CreateBulk(roster);
 
-                    var domainTeam = teamViewModel.ToDomain();
-                    domainTeam.Captain = players.First().Id;
+                    var team = teamViewModel.ToCreateTeamDto();
 
-                    _teamService.Create(domainTeam);
+                    var createdTeam = _teamService.Create(team);
 
-                    if (teamViewModel.Roster != null)
-                    {
-                        _teamService.UpdateRosterTeamId(players, domainTeam.Id);
-                    }
+                    _teamService.AddPlayers(new TeamId(createdTeam.Id), players.Select(x=>new PlayerId(x.Id)));
```

Comment:
Why do you make it in 2 steps? Can you pass roster when creating team?

Reply:
ok